### ISSUE TITLE
Number of buckets in info listing doesn't go down after removing a bucket

### ISSUE THREAD

**Issue by justinfenn** [minio/minio#12512]
The `mc admin info <alias>` command lists the total number of buckets in the output, but this seems to include deleted buckets.  The number isn't reduced after deleting a bucket.

## Expected Behavior

The number of buckets in the server info should match the number seen by an `mc ls` command or in the web interface.  I would also expect it to go down after removing a bucket.

## Current Behavior

The number of buckets in the server info output can be higher than the real number of buckets, and it never seems to go down after removing buckets.

## Possible Solution

The root cache entry's children don't seem to be removed [here](https://github.com/minio/minio/blob/f30c996d4840923e745a1a434b6fb890fcd35685/cmd/data-usage-cache.go#L310).  The separate cache entries for deleted buckets are removed, but it doesn't look like those are used for the info count.

## Steps to Reproduce (for bugs)

1. Create two buckets in a fresh instance:
```bash
$ mc mb local/test1
$ mc mb local/test2
```
2. Wait for usage cache to update and `mc admin info local` to show two buckets
3. Delete one of the buckets:
```bash
$ mc rb local/test1
```
4. Wait for usage cache to update again and observe that the bucket count hasn't changed

## Context

I use the info command to get a general sense of the server state and storage utilization.  Over a long time with lots of bucket creation and deletion, the summary can get significantly out of sync with reality.

## Regression

Not sure, but I've seen this happening for quite a while.

## Your Environment

* Version used (`minio --version`): 2021-06-14T01:29:23Z
* Server setup and configuration: MinIO is running in Docker with the FS backend
* Operating System and version (`uname -a`): Linux server 5.12.10-arch1-1 #1 SMP PREEMPT Thu, 10 Jun 2021 16:34:50 +0000 x86_64 GNU/Linux

**Comment by harshavardhana** [comment]
It doesn't go down right away @justinfenn - it is only when the scanner picks up the new changes.  This happens once every  1 minute. 

How long did wait?

**Comment by justinfenn** [comment]
Thanks for the quick response.  In the simple test, I confirmed the usage cache had been updated through logs (`_MINIO_SERVER_DEBUG=on`) and looking at the cache file's modified timestamp.  I've also seen it get pretty far out of sync over months of real-world usage.

**Comment by harshavardhana** [comment]
Is your backend drive a networked drive?

**Comment by klauspost** [comment]
Buckets are removed on start of every cycle. This is indicated with `"starting scanner cycle"` in the debug output.

Could you post the output zip file from `mc admin profile start --type=goroutines myminio&&sleep 60&&mc admin profile stop myminio`.

**Comment by justinfenn** [comment]
@harshavardhana On my server, the backend is a local ZFS dataset mounted as a Docker volume.  In the test setup, the backend is a local tmpfs.

@klauspost I have attached the profile output.  I'm not sure if I was supposed to do something while this ran.  I removed a bucket and checked the info output after the profile command finished and saw that the bucket count was unchanged.

[profile.zip](https://github.com/minio/minio/files/6664414/profile.zip)
